### PR TITLE
Add Antrea as Network Policy Provider

### PR DESCRIPTION
**Pull request by yasensim**
Adding Project Antrea in the list of the Network Policy Providers.

### CODE REVIEW

File: content/en/docs/tasks/administer-cluster/network-policy-provider/antrea-network-policy.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+## Deploying Antrea in EKS and GKE
+
+Antrea can be deployed in NetworkPolicy only mode to an EKS cluster or a GKE
+cluster, and enforce NetworkPolicies for the cluster.
+
+* To deploy Antrea in an EKS cluster, please refer to [the EKS installation guide](https://github.com/vmware-tanzu/antrea/blob/master/docs/eks-installation.md).
+* To deploy Antrea in a GKE cluster, please refer to [the GKE installation guide](https://github.com/vmware-tanzu/antrea/blob/master/docs/gke-installation.md).
+
+## Deploying Antrea with IPsec Encryption
```
Comment (abhiraut) [inline content/en/docs/tasks/administer-cluster/network-policy-provider/antrea-network-policy.md:46]: i think it is worth while adding a reference to the Antrea on Windows config, which also provides a NetworkPolicy implementation for Windows

File: content/en/docs/tasks/administer-cluster/declare-network-policy.md [hunk 2]
```diff
@@ -18,6 +18,7 @@ This document helps you get started using the Kubernetes [NetworkPolicy API](/do
 
 Make sure you've configured a network provider with network policy support. There are a number of network providers that support NetworkPolicy, including:
 
+* [Antrea](/docs/tasks/administer-cluster/network-policy-provider/antrea-network-policy)
```
Comment (sftim) [inline content/en/docs/tasks/administer-cluster/declare-network-policy.md:21]:
```suggestion
* [Antrea](/docs/tasks/administer-cluster/network-policy-provider/antrea-network-policy/)
```

File: content/en/docs/tasks/administer-cluster/network-policy-provider/antrea-network-policy.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+weight: 10
+---
+
+<!-- overview -->
+This page shows how to create an Antrea cluster on Kubernetes.
+For background on Project Antrea, read the [Introduction to Antrea](https://antrea.io/docs/).
+
+## {{% heading "prerequisites" %}}
+
+You need a Kubernetes cluster created on premisses or in Public Cloud, and the kubectl command-line tool must be configured to communicate with your cluster.
```
Comment (sftim) [inline content/en/docs/tasks/administer-cluster/network-policy-provider/antrea-network-policy.md:13]:
```suggestion
You need a Kubernetes cluster created on premises or in public cloud, and the kubectl command-line tool must be configured to communicate with your cluster.
```

File: content/en/docs/tasks/administer-cluster/network-policy-provider/antrea-network-policy.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+<!-- steps -->
+
+## Deploying Antrea with kubeadm
+
+When using `kubeadm` to create the Kubernetes cluster, passing
+`--pod-network-cidr=<CIDR Range for Pods>` to `kubeadm init` will enable
+`NodeIpamController`. Clusters created with kubeadm will always have
```
Comment (sftim) [inline content/en/docs/tasks/administer-cluster/network-policy-provider/antrea-network-policy.md:21]:
```suggestion
`NodeIpamController`. Clusters created with kubeadm always have
```

File: content/en/docs/tasks/administer-cluster/network-policy-provider/antrea-network-policy.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+`CNI` plugins enabled. Refer to
+[Creating a cluster with kubeadm](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm)
```
Comment (sftim) [inline content/en/docs/tasks/administer-cluster/network-policy-provider/antrea-network-policy.md:23]:
```suggestion
[Creating a cluster with kubeadm](/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm)
```

File: content/en/docs/tasks/administer-cluster/network-policy-provider/antrea-network-policy.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+for more information about setting up a Kubernetes cluster with `kubeadm`.
+
+To deploy the latest version of Antrea (built from the master branch), use the
+checked-in [deployment yaml](https://github.com/vmware-tanzu/antrea/blob/master/build/yamls/antrea.yml):
```
Comment (sftim) [inline content/en/docs/tasks/administer-cluster/network-policy-provider/antrea-network-policy.md:27]:
```suggestion
checked-in [deployment manifest](https://github.com/vmware-tanzu/antrea/blob/master/build/yamls/antrea.yml):
```

File: content/en/docs/tasks/administer-cluster/network-policy-provider/antrea-network-policy.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/vmware-tanzu/antrea/master/build/yamls/antrea.yml
+```
+
+## Deploying Antrea in Kind
+
+To deploy Antrea in a [Kind](https://github.com/kubernetes-sigs/kind) cluster,
+please refer to this [guide](https://github.com/vmware-tanzu/antrea/blob/master/docs/kind.md).
```
Comment (sftim) [inline content/en/docs/tasks/administer-cluster/network-policy-provider/antrea-network-policy.md:36]:
```suggestion
refer to [Deploying Antrea on a Kind cluster](https://github.com/vmware-tanzu/antrea/blob/master/docs/kind.md).
```

File: content/en/docs/tasks/administer-cluster/network-policy-provider/antrea-network-policy.md [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
+## Deploying Antrea in EKS and GKE
+
+Antrea can be deployed in NetworkPolicy only mode to an EKS cluster or a GKE
+cluster, and enforce NetworkPolicies for the cluster.
+
+* To deploy Antrea in an EKS cluster, please refer to [the EKS installation guide](https://github.com/vmware-tanzu/antrea/blob/master/docs/eks-installation.md).
+* To deploy Antrea in a GKE cluster, please refer to [the GKE installation guide](https://github.com/vmware-tanzu/antrea/blob/master/docs/gke-installation.md).
```
Comment (sftim) [inline content/en/docs/tasks/administer-cluster/network-policy-provider/antrea-network-policy.md:44]: A task page should focus as much as possible on _a_ task. It's OK to focus on the kind, or minikube, or GKE story - but that focus _is_ important. I'd remove this text.

**Review by sftim — CHANGES_REQUESTED** [review]
@yasensim would you be willing to update this into a _shorter_ page that directs readers to read the Antrea documentation for details? I'd expect to see links to pages inside https://antrea.io/docs/

File: content/en/docs/tasks/administer-cluster/network-policy-provider/antrea-network-policy.md [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+
+## Deploying Antrea with IPsec Encryption
+
+Antrea supports encrypting GRE tunnel traffic with IPsec. To deploy Antrea with
+IPsec encryption enabled, please refer to this [guide](https://github.com/vmware-tanzu/antrea/blob/master/docs/ipsec-tunnel.md).
```
Comment (sftim) [inline content/en/docs/tasks/administer-cluster/network-policy-provider/antrea-network-policy.md:49]:
```suggestion
 IPsec encryption enabled, please refer to [IPsec Encryption of Tunnel Traffic with Antrea](https://github.com/vmware-tanzu/antrea/blob/master/docs/ipsec-tunnel.md).
```

File: content/en/docs/tasks/administer-cluster/network-policy-provider/antrea-network-policy.md (hunk 8 above)
Comment (yasensim) [reply content/en/docs/tasks/administer-cluster/network-policy-provider/antrea-network-policy.md:44]: @sftim thanks fo your help. This makes sense. I will try to reduce the content.

**Comment by sftim** [comment]
Hi @yasensim 

Would you have time to revise this PR so it's in line with the [content guide](https://kubernetes.io/docs/contribute/style/content-guide/)?

**Comment by sftim** [comment]
/hold

As it stands, this PR has too much content about third-party products that are _not_ required to operate Kubernetes.

**Comment by yasensim** [comment]
Thanks @sftim , I will close this one and will send another one with following the guide.

**Comment by luolanzone** [comment]
Hi @yasensim  is this still on your radar?

**Pull request closed without merging**